### PR TITLE
[oneMKL][RNG] Samples clean up for RNG, add new sample.

### PR DESCRIPTION
**Pull request by aelizaro**
# Description

- W/a is added for reduce function (namespace changes from sycl::intel to sycl::ONEAPI::)

- European option pricing sample is cleaned and aligned with pi number evaluation sample

- Add new RNG sample: Multiple Simple Random Sampling without replacement

Fixes # (issue) 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Manually tests for Linux and Windows CPU and GPU with oneMKL beta08 and oneMKL beta09 functionality

- [x] Command Line

### CODE REVIEW

File: Libraries/oneMKL/random_sampling_without_replacement/sample.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "dependencies": [ "mkl" ],
+    "description": "Multiple simple random sampling without replacement with oneMKL random number generators.",
+    "languages": [ { "cpp": { "properties": { "projectOptions": [ { "projectType": "makefile" } ] } } } ],
+    "targetDevice": [ "CPU", "GPU" ],
+    "os": [ "linux", "windows" ],
+    "builder": [ "make" ],
+    "ciTests": {
+      "linux": [
+        {
+          "id": "random_sampling_without_replacement",
```
Comment (mkitez) [inline Libraries/oneMKL/random_sampling_without_replacement/sample.json:15]: IDs are not required if there's a single test object in the array, they are mostly to distinguish test executions if there are two or more

Comment (aelizaro) [reply Libraries/oneMKL/random_sampling_without_replacement/sample.json]: Hi @mkitez, I have removed "id"

Comment (mkitez) [reply Libraries/oneMKL/random_sampling_without_replacement/sample.json]: Thanks

File: Libraries/oneMKL/monte_carlo_european_opt/mc_european.cpp [hunk 2]
```diff
@@ -24,22 +24,31 @@
...
+
 // Default number of options
-#define N_OPT  2048
+static const auto n_opt_default = 2048;
 // Default number of independent samples
-#define N_SAMPLES 262144
+static const auto n_samples_default = 262144;
 
 // Initialization value for random number generator
-#define SEED        7777
```
Comment (JoeOster) [inline Libraries/oneMKL/monte_carlo_european_opt/mc_european.cpp:33]: in general, the variable names are not very descriptive and make it difficult to track

Comment (aelizaro) [reply Libraries/oneMKL/monte_carlo_european_opt/mc_european.cpp:33]: Hi @JoeOster, SEED is a common name for random number generators usage, I have followed the same approach for variables naming as in [the classic MKL cookbook sample](https://software.intel.com/content/www/us/en/develop/documentation/mkl-cookbook/top/using-the-monte-carlo-method-for-simulating-european-options-pricing.html#using-the-monte-carlo-method-for-simulating-european-options-pricing)

Comment (JoeOster) [reply Libraries/oneMKL/monte_carlo_european_opt/mc_european.cpp:33]: I wasn't being specific with SEED, rather was just a general comment about variables throughout.

Comment (aelizaro) [reply Libraries/oneMKL/monte_carlo_european_opt/mc_european.cpp:33]: ok, fixed most variables names in Monte Carlo european options sample.

**Review by JoeOster — APPROVED** [review]

**Pull request merged**